src/ncp/ncp_uart.cpp

```diff
@@ -287,6 +287,8 @@ void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat
     }
     va_end(args);
 
+    otNcpStreamWrite(0, reinterpret_cast<const uint8_t*>("\r\n"), 2);
```

Comment:
Not sure we need this.

Instead you can use `OPENTHREAD_CONFIG_LOG_SUFFIX` from `openthread-core-default-config.h` and define it to be `"\r\n"`.  The suffix will be added to end of all log lines (check `logging.hpp`).